app/Http/Controllers/Group/Workspace/Topic/BillSearchController.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Http\Controllers\Group\Workspace\Topic;
+
+use App\Http\Controllers\Controller;
+use App\Models\Group;
+use App\Models\LegiScan\Session;
+use App\Models\Topic;
+use App\Models\Workspace;
+use Carbon\Carbon;
+
+class BillSearchController extends Controller
+{
+    public function show(Group $group, Workspace $workspace, Topic $topic)
+    {
+        $sessionYears = Session::validYears();
+
+        return view('groups.workspaces.topics.bill-search')
+            ->withGroup($group)
+            ->withWorkspace($workspace)
+            ->withTopic($topic)
```

Comment:
Technically you can just pass `$topic` and derive everything else from there, right?

```
$topic
$topic->workspace
$topic->workspace->group
```

You _can_ do it this way, but I would advise in general just passing the primary model you require.

Reply:
I can do that, but in this case, why wouldn't my route just be `topic/{topic}` without being nested under group or workspace?

Reply:
No. I get why you would think that, but that's only because you're only thinking about this method. There's 7 CRUD methods per resource level, and any of the others may require one or both of the other models in the chain in order to function.

It's just that for this one method, you technically only need the topic in order to satisfy the view.

Reply:
Ok. For now, I will keep it in... because I'm lazy. I can always refactor it out later, if that's ok.